apps/e2e-test/src/logger/logger.ts

```diff
@@ -0,0 +1,50 @@
+import { CustomLogger } from './custom-logger';
+
+export class Logger {
+  private static instance: CustomLogger<unknown>;
+
+  private static logMap = new Map<string, number>([
+    ['trace', 1],
+    ['debug', 2],
+    ['info', 3],
+    ['warn', 4],
+    ['error', 5],
+    ['fatal', 6],
+  ]);
+
+  public static getInstance(logLevel = 'info'): CustomLogger<unknown> {
+    if (!Logger.instance) {
+      Logger.instance = Logger.createLogger(logLevel);
+    }
+    return Logger.instance;
+  }
+
+  private static createLogger(logLevel = 'info'): CustomLogger<unknown> {
+    const settings = {
+      minLevel: Logger.getMinLevel(logLevel),
+      prettyLogTemplate: '{{dd}}-{{mm}}-{{yyyy}} {{hh}}:{{MM}}:{{ss}}:{{ms}} :: {{logLevelName}} ::',
+      prettyLogStyles: {
+        logLevelName: {
+          '*': ['bold', 'black', 'bgWhiteBright', 'dim'],
+          SILLY: ['bold', 'white'],
+          TRACE: ['bold', 'whiteBright'],
+          DEBUG: ['bold', 'green'],
+          INFO: ['bold', 'blue'],
+          WARN: ['bold', 'yellow'],
+          ERROR: ['bold', 'red'],
+          FATAL: ['bold', 'redBright'],
+          STEP: ['bold', 'green'],
+          SCENARIO: ['bold', 'green'],
```

Comment:
They don't quite tie up with what is defined in `logMap`. Do we need so many levels?

Reply:
Updated the logmap. This logger is warped around the tslog package hence provided all the log levels provided by tslog.

STEP and SCENARIO log level are the custom log levels internally used by the test framework and cannot be specified by the in test config by the user.